src/kemal/route_handler.cr

```diff
@@ -49,4 +42,14 @@ class Kemal::RouteHandler < HTTP::Handler
     node = radix_path method, path
     @tree.add node, route
   end
+
+  class RouteNotFoundException < Exception
+    def initialize(method, path)
+      super "Requested path: '#{method}:#{path}' was not found."
+    end
+  end
+
+  class NilPathException < Exception
```

Comment:
Oh ok so you meant only the last exception was necessary, sorry I misunderstood. I just tried to replicate the previous behavior as closely as possible. Before an exception was raised by `.not_nil!` above so I just thought a clearer exception was better, I'll remove it if it's unecessary.